base/ApplicationSupport.F90

```diff
@@ -39,8 +39,8 @@ subroutine MAPL_Initialize(unusable,comm,logging_config,rc)
       call initialize_pflogger(comm=comm_world,logging_config=logging_configuration_file,rc=status)
       _VERIFY(status)
 #endif
-      call start_global_profiler(comm=comm_world,rc=status)
-      _VERIFY(status)
+      call initialize_global_time_profiler(comm=comm_world)
+      call start_global_time_profiler()
```

Comment:
No - we need a pure initialize() and finalize() procedure in the profiling package.  Those can start/stop the global timer.   This gives us maximum flexibility for the future and also makes the packages consistent.

Reply:
We may need a counter class to trace how many profilers are created and the last one releases the types and op ?

Reply:
That's what the finalize() procedure is for.  It is up to the user to call it "after" all the profilers are done.    I don't think we need a counter (or profiler set) for this purpose, but could be considered for later.  For now, just do the simple thing which is all-but-equivalent to what you've done above.  It allows the profiler to evolve without changing interfaces.

Reply:
Do you mean put start and stop inside the initialize and finalize? We probably can't

Reply:
No.  Create 2 procedures for the profiling package:   initialize() and finalize().      Initialize() will call initialize_global_time_profiler(...)  and finalize() will call finalize_global_time_profiler().   Then in Application support just call initialize() and finalize().  This one level of indirection lets us add complexity later in the profiling package without having to touch application_support() again.

Reply:
Where should we put these two subroutines?

Reply:
https://github.com/GEOS-ESM/MAPL/blob/main/profiler/MAPL_Profiler.F90

Reply:
I have pushed my changes